packages/headless/src/features/recent-results/recent-results-actions.ts

```diff
@@ -0,0 +1,58 @@
+import {createAction} from '@reduxjs/toolkit';
+import {ArrayValue, NumberValue, RecordValue} from '@coveo/bueno';
+import {validatePayload} from '../../utils/validate-payload';
+import {Result} from '../../api/search/search/result';
+import {
+  resultPartialDefinition,
+  validateResultPayload,
+} from '../analytics/analytics-utils';
+
+export interface RegisterRecentResultsCreatorPayload {
+  /**
+   * The recent results viewed by the user prior to instantiating the controller.
+   */
+  results: Result[];
+  /**
+   * The maximum number of queries to retain in the list.
+   */
+  maxLength: number;
+}
+
+const registerRecentResultsPayloadDefinition = {
+  results: new ArrayValue({
+    required: true,
+    each: new RecordValue({values: resultPartialDefinition}),
+  }),
+  maxLength: new NumberValue({required: true, min: 1, default: 10}),
+};
+
+/**
+ * Initialize the `recentResults` state.
+ * @param payload (RegisterRecentResultsCreatorPayload) The initial state and options.
+ */
+export const registerRecentResults = createAction(
+  'recentResults/registerRecentResults',
```

Comment:
Do you think that putting all the actions magic strings in a const object might help with readability ?

Reply:
This is the pattern that's used everywhere else in headless and I'd prefer to stay consistent. The string isn't referenced elsewhere so this magic string is only declared here.